### ISSUE TITLE
`jit pull` fails: upstream branch 'refs/heads/main' not stored as a remote-tracking branch

### ISSUE THREAD

**Issue by barrettj12** [barrettj12/jit#45]
```
$ jit rebase
Pulling branch "main"...
Error: pulling branch "main": getting upstream for branch "main": fatal: upstream branch 'refs/heads/main' not stored as a remote-tracking branch

exit status 128
```

**Comment by barrettj12** [comment]
Same error with `jit pull`:
```
$ jit pull main
pulling branch "main"...
Error: getting upstream for branch "main": fatal: upstream branch 'refs/heads/main' not stored as a remote-tracking branch

exit status 128
```

**Comment by barrettj12** [comment]
The problem was that `git config remote.<remote>.fetch` wasn't set - it should be set to `+refs/heads/*:refs/remotes/<remote>/*`.

I think this is automatically done by Jit when cloning or creating remotes - this should be tested though.

Maybe we should have some logic that checks for this error, and sets git config if the pull fails?

**Comment by barrettj12** [comment]
Fixed by #48
